### PR TITLE
Implemented: logic to edit data #22c8vgh

### PR DESCRIPTION
**Pull request by disha1202**

### CODE REVIEW

File: src/store/modules/order/OrderState.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export default interface OrderState {
+    order: {
```
Comment (adityasharma7) [inline src/store/modules/order/OrderState.ts:2]: https://github.com/hotwax/import/pull/5/files#r815722811

File: src/store/modules/order/actions.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+
+const actions: ActionTree<OrderState, RootState> = {
+  updateOrderList ({ commit }, payload) {
+    commit(types.ORDER_LIST_UPDATED,  payload );
+  },
+  modifyCsv ({ commit }, payload) {
+    commit(types.MODIFY_CSV, payload);
+  },
+  async groupProducts ({commit}, csv: any) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:16]: I this should a getter instead

File: src/store/modules/order/actions.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import OrderState from './OrderState'
+import * as types from './mutation-types'
+import router from '@/router'
+
+
+const actions: ActionTree<OrderState, RootState> = {
+  updateOrderList ({ commit }, payload) {
+    commit(types.ORDER_LIST_UPDATED,  payload );
+  },
+  modifyCsv ({ commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:13]: We will be performing updates on order item list only. The state will not update any CSV, but order list

File: src/store/modules/order/actions.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    commit(types.MODIFY_CSV, payload);
+  },
+  async groupProducts ({commit}, csv: any) {
+    const productIds = csv.map((item: any) => {
+      return item.shopifyproductSKU
+    })
+    const viewSize = process.env.VUE_APP_VIEW_SIZE;
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:20]: I think value should be productIds length

File: src/store/modules/order/actions.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      viewIndex,
+      productIds
+    }
+    const resp = await store.dispatch("product/fetchProducts", payload);
+      csv = csv.map((item: any) => {
+        const product = resp.data.response.docs.find((product: any) => {
+          if(item.shopifyproductSKU == product.internalName)
+          return product;
+        })
+        console.log(product);
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:33]: It seems we missed a console log

File: src/store/modules/order/actions.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        item.groupId = product.groupId;
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:34]:
```suggestion
        item.parentProductId = product.groupId;
```

File: src/store/modules/order/actions.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        item.internalName = product.internalName; 
+        item.parentProductName = product.parentProductName;
+        item.imageUrl = product.mainImageUrl;
+        item.isNewProduct = false;
+        return item;
+    })
+    await store.dispatch('order/updateOrderList', csv);
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:41]: Improve variable name: csv  
We will only deal with order list

File: src/store/modules/order/mutations.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { MutationTree } from 'vuex'
+import OrderState from './OrderState'
+import * as types from './mutation-types'
+
+const mutations: MutationTree <OrderState> = {
+  [types.ORDER_LIST_UPDATED] (state, payload) {
+    state.order.originalCsv = payload;
+    console.log(payload);
```
Comment (adityasharma7) [inline src/store/modules/order/mutations.ts:8]: console log missed

File: src/views/PurchaseOrder.vue [hunk 9]
```diff
@@ -66,23 +78,58 @@ export default defineComponent({
...
       getFile(event) {
-        const file = ref(null);
         this.file = event.target.files[0];
         this.parseFile();
       },
       async parseFile(){
         await parseCsv(this.file).then(res => {
           this.content = res;
-          console.log(res);
+          // console.log(res);
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:99]: Missed console log

File: src/views/PurchaseOrder.vue [hunk 10]
```diff
@@ -66,23 +78,58 @@ export default defineComponent({
...
         })
       },
-    },
-    
+      mapFields() {
+        // console.log(this.csvParsed);
+        
+        this.csvParsed = this.content.map(item => {
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:105]: order item list

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/OrderState.ts (hunk 1 above)
Comment (disha1202) [reply src/store/modules/order/OrderState.ts:2]: Updated the order state

File: src/store/modules/order/actions.ts (hunk 2 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: In this action we are fetching product information and storing in order state.
Can we still add it in getters?

File: src/store/modules/order/actions.ts (hunk 3 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Updated the code.

File: src/store/modules/order/actions.ts (hunk 4 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Made the requested change.

File: src/store/modules/order/actions.ts (hunk 6 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Made the requested change

File: src/store/modules/order/actions.ts (hunk 7 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Improved variable name

File: src/store/modules/order/mutations.ts (hunk 8 above)
Comment (disha1202) [reply src/store/modules/order/mutations.ts]: Removed

File: src/views/PurchaseOrder.vue (hunk 9 above)
Comment (disha1202) [reply src/views/PurchaseOrder.vue]: Removed.

File: src/views/PurchaseOrder.vue (hunk 10 above)
Comment (disha1202) [reply src/views/PurchaseOrder.vue]: Updated.

File: src/store/modules/order/actions.ts [hunk 11]
```diff
@@ -0,0 +1,43 @@
+import { ActionTree } from 'vuex'
+import store from '@/store'
+import RootState from '@/store/RootState'
+import OrderState from './OrderState'
+import * as types from './mutation-types'
+import router from '@/router'
+
+
+const actions: ActionTree<OrderState, RootState> = {
+  orderListOriginal ({ commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:10]: I think this is not needed

File: src/store/modules/order/index.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+import { Module } from 'vuex'
+import actions from './actions'
+import mutations from './mutations'
+import OrderState from './OrderState'
+import RootState from '../../RootState'
+
+const productModule: Module<OrderState, RootState> = {
+  namespaced: true,
+  state: {
+    order: {
```
Comment (adityasharma7) [inline src/store/modules/order/index.ts:11]:
```
      list: {
        items: [],
        original: []
      }
```

File: src/views/OrderDetail.vue [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+    computed: {
+      ...mapGetters({
+        orderItems: 'order/getOrderItems',
+        getProduct: 'product/getProduct',
+      }),
+    },
+    data() {
+      return {
+        numberOfDays: 0,
+        numberOfpieces: 0,
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:134]:
```suggestion
        numberOfPieces: 0,
```

File: src/views/OrderDetail.vue [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+        catalog: "",
+        listOfCheckedProducts: [] as any,
+        originalCsv: {}
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:137]: Improve variable name

File: src/views/OrderDetail.vue [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+      }
+    },
+    methods: {
+      onChange(productSku: string) {
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:141]: Improve method name. It should reflect the operation

File: src/views/PurchaseOrder.vue [hunk 16]
```diff
@@ -66,23 +78,53 @@ export default defineComponent({
...
-          console.log(res);
         })
       },
-    },
-    
+      mapFields() {
+        this.orderItemsList = this.content.map(item => {
+          const orderItem = {
+          orderId: [],
+          shopifyproductSKU: [],
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:104]:
```suggestion
          shopifyProductSKU: [],
```

File: src/views/PurchaseOrder.vue [hunk 17]
```diff
@@ -66,23 +78,53 @@ export default defineComponent({
...
+          shopifyproductUPC: [],
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:105]:
```suggestion
          shopifyProductUPC: [],
```

File: src/store/modules/order/actions.ts [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+    commit(types.ORDER_LIST_ORIGINAL,  payload );
+  },
+  async groupProducts ({commit}, orderItems: any) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:13]: Improve method name

File: src/store/modules/order/actions.ts [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+    const payload = {
+      viewSize,
+      viewIndex,
+      productIds
+    }
+    const resp = await store.dispatch("product/fetchProducts", payload);
+    orderItems = orderItems.map((item: any) => {
+        const product = resp.data.response.docs.find((product: any) => {
+          if(item.shopifyproductSKU == product.internalName)
+          return product;
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:28]: I think if is not needed
```suggestion
          return item.shopifyproductSKU == product.internalName;
```

File: src/views/OrderDetail.vue [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+    <ion-content :fullscreen="true">
+      <div class="header">
+        <div class="search">
+          <ion-searchbar> </ion-searchbar>
+        </div> 
+
+        <div class="filters">
+          <ion-item>
+            <ion-label>{{ $t("Buffer days") }}</ion-label>
+            <ion-input v-model="numberOfDays" type="text" placeholder="all items" />
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:29]: Placeholder should be internationalised

File: src/views/OrderDetail.vue [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+          </ion-item>
+          <ion-button expand="block" fill="outline" @click="apply">{{ $t("Apply") }}</ion-button>
+        </div> 
+      </div>  
+
+      <div v-for="id in getGroupList(orderItems)" :key="id" >
+        <div v-for="item in getGroupItems(id, orderItems)" :key="item">
+          <div class="list-header" >
+            <ion-label>{{ item.parentProductName }}</ion-label>
+            <ion-chip >
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:50]: You could use self closing `<div />` for empty space

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/actions.ts [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+        })
+        item.parentProductId = product.groupId;
+        item.internalName = product.internalName; 
+        item.parentProductName = product.parentProductName;
+        item.imageUrl = product.mainImageUrl;
+        item.isNewProduct = false;
+        return item;
+    })
+    await store.dispatch('order/orderListOriginal', orderItems);
+    router.push({
```
[the remark this answers is not in the document]
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Moved to the specific component.

File: src/store/modules/order/index.ts (hunk 12 above)
Comment (disha1202) [reply src/store/modules/order/index.ts]: Updated order state

File: src/views/OrderDetail.vue (hunk 13 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Made the requested change.

File: src/views/OrderDetail.vue (hunk 14 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Improved.

File: src/views/OrderDetail.vue (hunk 15 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Improved method name.

File: src/views/PurchaseOrder.vue (hunk 16 above)
Comment (disha1202) [reply src/views/PurchaseOrder.vue]: Made the requested change.

File: src/views/PurchaseOrder.vue (hunk 17 above)
Comment (disha1202) [reply src/views/PurchaseOrder.vue]: Made the requested change.

File: src/store/modules/order/actions.ts (hunk 18 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Improved method name.

File: src/store/modules/order/actions.ts (hunk 19 above)
Comment (disha1202) [reply src/store/modules/order/actions.ts]: Made the requested change.

File: src/views/OrderDetail.vue (hunk 20 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Internationalised placeholder text.

File: src/views/OrderDetail.vue (hunk 21 above)
Comment (disha1202) [reply src/views/OrderDetail.vue]: Used self closing <div />  for empty space.

File: src/store/modules/order/actions.ts [hunk 23]
```diff
@@ -0,0 +1,37 @@
+import { ActionTree } from 'vuex'
+import store from '@/store'
+import RootState from '@/store/RootState'
+import OrderState from './OrderState'
+import * as types from './mutation-types'
+import router from '@/router'
+
+
+const actions: ActionTree<OrderState, RootState> = {
+  async orderListUpdated ({commit}, orderItems) {
```
Comment (adityasharma7) [inline src/store/modules/order/actions.ts:10]:
```suggestion
  async updatedOrderList ({commit}, orderItems) {
```

File: src/store/modules/order/mutation-types.ts [hunk 24]
```diff
@@ -0,0 +1,3 @@
+export const SN_ORDER = 'order'
+export const ORDER_LIST_UPDATED = SN_ORDER + '/LIST_UPDATED'
+export const ORDER_ITEMS_UPDATED = SN_ORDER + '/ITEMS_UPDATED'
```
Comment (adityasharma7) [inline src/store/modules/order/mutation-types.ts:3]: I think this is not used anywhere. We could remove

File: src/views/OrderDetail.vue [hunk 25]
```diff
@@ -0,0 +1,217 @@
...
+          this.listOfCheckedProducts.splice(this.listOfCheckedProducts.indexOf(productSku, 1));
+        }
+        else {
+          this.listOfCheckedProducts.push(productSku);
+        }
+      },
+      apply() {
+        this.ordersList.items.map((item: any) => {
+          if (this.listOfCheckedProducts.includes(item.shopifyProductSKU)) {
+            item.quantityOrdered -= this.numberOfPieces;
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:144]: I think here we should only updates values changed

File: src/views/OrderDetail.vue [hunk 26]
```diff
@@ -0,0 +1,217 @@
...
+        })
+        this.store.commit('order/order/ITEMS_UPDATED', this.ordersList.items);
+      },
+      getGroupList (items: any) {
+        return Array.from(new Set(items.map((ele: any) => ele.parentProductId)))
+      },
+      getGroupItems(parentProductId: any, items: any) {
+        return items.filter((item: any) => item.parentProductId == parentProductId)
+      },
+      checkAllProducts() {
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:158]:
```suggestion
      selectAllItems() {
```

File: src/views/OrderDetail.vue [hunk 27]
```diff
@@ -0,0 +1,217 @@
...
+        this.ordersList.items.forEach((item: any) => {
+          this.checkProducts(item.shopifyProductSKU);
+        })
+      },
+      checkGroupedProducts(parentProductId: any){
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:163]:
```suggestion
      selectParentProduct(parentProductId: any){
```

File: src/views/PurchaseOrder.vue [hunk 28]
```diff
@@ -20,34 +20,46 @@
           <ion-note>{{ $t("Select the column index for the following information in the uploaded CSV.") }}</ion-note>
           <ion-item>
               <ion-label>{{ $t("Order ID") }}</ion-label>
-              <ion-select placeholder="Select"></ion-select>
+              <ion-select v-if="content.length" placeholder="Select" v-model="orderIdField">
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:23]: Internationalise placeholder text

File: src/views/PurchaseOrder.vue [hunk 29]
```diff
@@ -66,23 +78,56 @@ export default defineComponent({
...
+        }
+          orderItem.orderId = item[this.orderIdField];
+          orderItem.shopifyProductSKU = item[this.productSkuField];
+          orderItem.shopifyProductUPC = item[this.productUpcField];
+          orderItem.arrivalDate = item[this.dateField];
+          orderItem.quantityOrdered = item[this.quantityField];
+          return orderItem
+        })
+        this.store.dispatch('order/orderListUpdated', this.orderItemsList);
+        console.log(this.orderItemsList);
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:117]: It seems there is console log left

File: src/store/modules/order/getters.ts [hunk 30]
```diff
@@ -0,0 +1,10 @@
+import { GetterTree } from "vuex";
+import OrderState from "./OrderState";
+import RootState from "../../RootState";
+
+const getters: GetterTree<OrderState, RootState> = {
+  getOrdersList(state){
```
Comment (adityasharma7) [inline src/store/modules/order/getters.ts:6]:
```suggestion
  getOrder(state){
```

File: src/views/OrderDetail.vue [hunk 31]
```diff
@@ -0,0 +1,217 @@
...
+        </ion-buttons>
+        
+        
+      </ion-toolbar>
+    </ion-header>
+
+    <ion-content :fullscreen="true">
+      <div class="header">
+        <div class="search">
+          <ion-searchbar> </ion-searchbar>
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:23]:
```suggestion
          <ion-searchbar />
```

File: src/views/OrderDetail.vue [hunk 32]
```diff
@@ -0,0 +1,217 @@
+<template>
+  <ion-page>
+    <ion-header :translucent="true">
+      <ion-toolbar>
+        <ion-back-button slot="start" default-href="/home" />
+        <ion-title>{{ $t("PO External Order ID") }}</ion-title>
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:6]: Internationalise text and check for other text as well

File: src/views/OrderDetail.vue [hunk 33]
```diff
@@ -0,0 +1,217 @@
...
+        else {
+          this.listOfCheckedProducts.push(productSku);
+        }
+      },
+      apply() {
+        this.ordersList.items.map((item: any) => {
+          if (this.listOfCheckedProducts.includes(item.shopifyProductSKU)) {
+            item.quantityOrdered -= this.numberOfPieces;
+            item.arrivalDate = DateTime.fromFormat(item.arrivalDate, "D").plus({ days: this.numberOfDays }).toFormat('MM/dd/yyyy');
+            if (this.catalog == "Preorder") item.isNewProduct = true
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:146]:
```suggestion
            item.isNewProduct = this.catalog == "Preorder"
```

File: src/views/OrderDetail.vue [hunk 34]
```diff
@@ -0,0 +1,217 @@
...
+      },
+      getGroupList (items: any) {
+        return Array.from(new Set(items.map((ele: any) => ele.parentProductId)))
+      },
+      getGroupItems(parentProductId: any, items: any) {
+        return items.filter((item: any) => item.parentProductId == parentProductId)
+      },
+      checkAllProducts() {
+        this.ordersList.items.forEach((item: any) => {
+          this.checkProducts(item.shopifyProductSKU);
```
Comment (adityasharma7) [inline src/views/OrderDetail.vue:160]: If one of them is already selected this will unselect it, but it should select all

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/order/actions.ts [hunk 35]
```diff
@@ -0,0 +1,40 @@
...
+        })
+        item.parentProductId = product.groupId;
+        item.internalName = product.internalName; 
+        item.parentProductName = product.parentProductName;
+        item.imageUrl = product.mainImageUrl;
+        item.isNewProduct = false;
+        return item;
+    })
+    const original = JSON.parse(JSON.stringify(items))
+    commit(types.ORDER_LIST_UPDATED, { items: items, original: original });
```
Comment (ymaheshwari1) [inline src/store/modules/order/actions.ts:34]:
```suggestion
    commit(types.ORDER_LIST_UPDATED, { items, original });
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**